test/specs/token.test.mjs

```diff
@@ -0,0 +1,230 @@
+
+      assert.strictEqual(typeof balanceData, 'string');
+      assert.strictEqual(typeof balanceTag, 'string');
+      assert.strictEqual(balanceData, balanceTag);
+      assert.strictEqual(Number.isInteger(parseInt(balanceData)), true);
+      assert.strictEqual(typeof ticker, 'string');
+      assert.strictEqual(targetAccountTag?.value ?? STUB_ADDRESS, account);
+      assert(ref, 'Not a reply message');
+    }
+  });
+
+  it('should get the balances', async () => {
+    /**
+     * Assert each balance is a string and int,
+     */
+    const result = await handle({
+      Tags: [{ name: 'Action', value: 'Balances' }],
+    });
+
+    const tags = result.Messages[0].Tags;
+    const ref = tags.find((t) => t.name === 'X-Reference')?.value;
+    assert(ref, 'Not a reply message');
+
+    for (const bal of Object.values(JSON.parse(result.Messages[0].Data))) {
+      assert.strictEqual(typeof bal, 'string');
+      assert.strictEqual(Number.isInteger(parseInt(bal)), true);
+    }
+  });
+
+  it('should transfer', async () => {
+    /**
+     * Check as:
+     * - Owner (sufficient balance)
+     * - Not owner (will send error message)
+     */
+    const recipient = ''.padEnd(43, 'recipient-1');
+    const result = await handle({
+      Tags: [
+        { name: 'Action', value: 'Transfer' },
+        { name: 'Recipient', value: recipient },
+      ],
+    });
+
+    const creditNotice = result.Messages.find(({ Tags }) =>
+      Tags.find((t) => t.name === 'Action' && t.value === 'Credit-Notice'),
+    );
+    assert(creditNotice, 'missing credit notice');
+    const debitNotice = result.Messages.find(({ Tags }) =>
+      Tags.find((t) => t.name === 'Action' && t.value === 'Debit-Notice'),
+    );
+    const ref = debitNotice.Tags.find((t) => t.name === 'X-Reference')?.value;
+    assert(ref, 'Credit notice is not a reply message');
+    assert(debitNotice, 'Missing debit notice');
+
+    const infoRes = await handle(
+      {
+        Tags: [{ name: 'Action', value: 'Info' }],
+      },
+      result.Memory,
+    );
+    const ownerAfterTransfer = infoRes.Messages[0].Tags.find(
+      (t) => t.name === 'Owner',
+    )?.value;
+    assert.strictEqual(
+      ownerAfterTransfer,
+      recipient,
+      'Owner after transfer not equal to recipient',
+    );
+
+    // non owner transfer
+    const badResult = await handle({
+      From: recipient,
+      Owner: recipient,
+      Tags: [
+        { name: 'Action', value: 'Transfer' },
+        { name: 'Recipient', value: STUB_ADDRESS },
+      ],
+    });
+
+    // should return an insufficient funds error
+    const errorMessage = badResult.Messages[0];
+    const errorRef = debitNotice.Tags.find(
+      (t) => t.name === 'X-Reference',
+    )?.value;
+    assert(errorRef, 'Error notice is not a reply message');
+    const errorTag = errorMessage.Tags.find((t) => t.name === 'Error');
+    assert.strictEqual(errorTag.value, 'Insufficient Balance!');
```

Comment:
Since this is mainly for integration with other platforms i wasn't certain on how granular to go here, so decided on exact error messages being enforced.